spec/features/order/user_places_order_spec.rb

```diff
@@ -30,15 +30,20 @@
 
       select("HSX", from: "machine_id")
 
-      click_button "Submit"
+      click_on "Submit"
 
       select("Custom Sequencing", from: "order_cost_object_id")
 
       fill_in "Comment", with: "Need this asap"
-      fill_in "order[kit_orders_attributes][0][amount]", with: "1"
-      fill_in "order_needed_by", with: "07/10/2015"
-      click_button "Submit"
+      fill_in "order_kit_orders_attributes_0_amount", with: "1"
+
+      fill_in "order_needed_by", with: @order.needed_by
+
+
+      find('.button').click
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.